zfs/zfs.go

```diff
@@ -799,18 +799,9 @@ func (a ZFSSendArgsUnvalidated) validateEncryptionFlagsCorrespondToResumeToken(c
 		return gen.fmt("resume token `toguid` != expected: %v != %v", t.ToGUID, a.To.GUID)
 	}
 
-	if a.Encrypted.B {
-		if !(t.RawOK && t.CompressOK) {
-			return ZFSSendArgsResumeTokenMismatchEncryptionNotSet.fmt(
-				"resume token must have `rawok` and `compressok` = true but got %v %v", t.RawOK, t.CompressOK)
-		}
-		// fallthrough
-	} else {
-		if t.RawOK || t.CompressOK {
-			return ZFSSendArgsResumeTokenMismatchEncryptionSet.fmt(
-				"resume token must not have `rawok` or `compressok` set but got %v %v", t.RawOK, t.CompressOK)
-		}
-		// fallthrough
+	if a.Encrypted.B && !(t.RawOK && t.CompressOK) {
```

Comment:
```suggestion
	if (a.Encrypted.B || a.Raw) && !(t.RawOK && t.CompressOK) {
```
should address the plaintext exfiltration issue imo.

@problame do we want/need to also check that -L and -e match? My gut feeling is no.

Do we need to rename/duplicate ZFSSendArgsResumeTokenMismatchEncryptionNotSet to ZFSSendArgsResumeTokenMismatchRawNotSet?

Reply:
Your proposed change returns the error unnecessarilty for the regular case:

```
token
  rawok = true
  compressor = false
config
  encrypted = false
  raw = true
  compress = false
```

Reply:
Re gut feeling: I don't think the other flags are relevant for security, but consider the following use case:
* new user sets up zrepl without any `send` options
* starts initial replication
* thinks "that'll take too long", interrupts initial replication
  * zfs saves resume token with (`compress = false, large_blocks = false`)
* user enables those features in the config, restarts daemon
* user expects that transfer will use `compress` and `large_blocks` now; they probably don't know and shouldn't care about the intricacies of the resume token

Reply:
> Your proposed change returns the error unnecessarilty for the regular case:

why is this different from how `encrypted: true` behaves, i.e. why is compression requested on top of `-w` (raw) with `encrypted` but not `raw`? Those configs should produce the same send flags and tokens imo.

Reply:
> why is compression requested on top of -w (raw) with encrypted but not raw?

That's just how it is with zfs, i.e., if you specify `send --raw` then the resume token is `rawok=true; compressor=true`

Reply:
(that is, for encrypted datasets. not sure about unencrypted datasets)